lmnet/lmnet/datasets/dataset_iterator.py

```diff
@@ -220,16 +221,22 @@ class DatasetIterator:
     available_subsets = ["train", "train_validation_saving", "validation"]
 
     """docstring for DatasetIterator."""
-    def __init__(self, dataset, enable_prefetch=False, seed=0):
+    def __init__(self, dataset, enable_prefetch=False, seed=0, batch_pre_fetch=3, num_parallel_process=1):
```

Comment:
About `num_parallel_process=1` , It is better to be number of CPU core as default. Or as thinking all most core has double logical core (using hyper-threading), to be half number of `cpu_count()` as default. 🤔

Reply:
@ruimashita San:
I prefer half of the `cpu_count()` as it will ensure having some free cpu resources for system processes or others even for the lower end machine having 2 or 4 cores.